src/components/OfflineIndicator.jsx

```
@@ -123,7 +123,7 @@
       if (networkCleanup) networkCleanup();
       if (syncCleanup) syncCleanup();
     };
-  }, []); // mount-only
+  }, [setupNetworkListeners]); // include setupNetworkListeners in dependencies
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)